examples/pick_mode.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/python3
+
+# Example of reading the available modes, and picking one with
+# the highest framerate and a raw bit depth of at least 10
+
+from picamera2 import Picamera2
+from picamera2.encoders import H264Encoder, Quality
+from picamera2.outputs import FfmpegOutput
+import time
+
+
+picam2 = Picamera2()
+
+available_modes = picam2.sensor_modes
+min_bit_depth = 10
+available_modes = list(filter(lambda x: (x["bit_depth"] >= min_bit_depth), available_modes))
+available_modes.sort(key=lambda x: x["fps"], reverse=True)
+[print(i) for i in available_modes]
+chosen_mode = available_modes[0]
+
+picam2.video_configuration = picam2.create_video_configuration(
+    raw={"size": chosen_mode["size"], "format": chosen_mode["format"].format}
+)
+picam2.configure("video")
+
+encoder = H264Encoder()
+output = FfmpegOutput('test.mp4')
+
+# Set the fps
+fps = int(chosen_mode["fps"])
```

Comment:
I was interested by the integer rounding here, but indeed it seems to be necessary. Looking at [controls.py](https://github.com/raspberrypi/picamera2/blob/main/picamera2/controls.py#L11), I use `//` to convert to convert the framerate to a duration but didn't realise that this can return a float (e.g. `1 // 1.0` returns `1.0`, not `1`). Oops.

I think it would be good to fix that up because requiring the framerate to be an integer is really not right (you couldn't ask for a frame rate of 1 frame per 2 seconds, for example)! Could I maybe ask you to add an extra first commit to this PR which resolves that problem in controls.py? Then the int cast here can be removed.